core-bundle/contao/models/PageModel.php

```diff
@@ -1171,7 +1171,10 @@ public function getPreviewUrl($strParams=null)
 		$baseUrl = $context->getBaseUrl();
 
 		// Add the preview script
-		$context->setBaseUrl(rtrim(\dirname($baseUrl), '/') . $previewScript);
+		if (false === strpos($baseUrl, $previewScript))
+		{
+			$context->setBaseUrl(rtrim($baseUrl, '/') . $previewScript);
+		}
```

Comment:
```suggestion
		$context->setBaseUrl(preg_replace('(/[^/]*$)', '', $baseUrl) . $previewScript);
```
Simpler alternative?

Reply:
🦊